### PR TITLE
Updating SDK reference. Updating extensions to use new builder model, configuration options and attributes

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: src/ExtensionsSample/Program.cs [hunk 1]
```diff
@@ -28,25 +28,23 @@ public static async Task Main(string[] args)
 
             var builder = new HostBuilder()
                .UseEnvironment("Development")
-               .ConfigureWebJobsHost(o =>
+               .ConfigureWebJobs(webJobsBuilder =>
```
Comment (mathewc) [inline src/ExtensionsSample/Program.cs]: We need a stable HostId workaround as I had below, for this sample to run. Please run and verify.

File: src/WebJobs.Extensions.CosmosDB/Config/CosmosDBWebJobsBuilderExtensions.cs [hunk 2]
```diff
@@ -11,40 +12,33 @@ namespace Microsoft.Extensions.Hosting
     /// <summary>
     /// Extension methods for CosmosDB integration.
     /// </summary>
-    public static class CosmosDBHostBuilderExtensions
+    public static class CosmosDBWebJobsBuilderExtensions
```
Comment (mathewc) [inline src/WebJobs.Extensions.CosmosDB/Config/CosmosDBWebJobsBuilderExtensions.cs]: I just updated all the SDK core extension names as well https://github.com/Azure/azure-webjobs-sdk-extensions/pull/452

File: src/WebJobs.Extensions.MobileApps/Config/MobileAppsWebJobsBuilderExtensions.cs [hunk 3]
```diff
@@ -11,41 +12,33 @@ namespace Microsoft.Extensions.Hosting
...
-            builder.AddExtension<MobileAppsExtensionConfigProvider>();
-            builder.ConfigureServices(services =>
-             {
-                 services.AddSingleton<IMobileServiceClientFactory, DefaultMobileServiceClientFactory>();
-                 services.AddOptions<MobileAppsOptions>()
-                     .Configure<IConfiguration>((options, config) =>
-                     {
-                         config.GetSection(ConfigSectionName)?.Bind(options);
-                     });
-             });
```
Comment (mathewc) [inline src/WebJobs.Extensions.MobileApps/Config/MobileAppsWebJobsBuilderExtensions.cs]: Nice!

File: src/WebJobs.Extensions.MobileApps/MobileAppsWebJobsStartup.cs [hunk 4]
```diff
@@ -5,13 +5,13 @@
 using Microsoft.Azure.WebJobs.Hosting;
 using Microsoft.Extensions.Hosting;
 
-[assembly: WebJobsStartup(typeof(MobileAppsWebJobsStartup))]
+[assembly: WebJobsStartup(typeof(MobileAppsWebJobsStartup), "Mobile Apps")]
```
Comment (mathewc) [inline src/WebJobs.Extensions.MobileApps/MobileAppsWebJobsStartup.cs]: I think we agreed that this name can include spaces, however for all the SDK core startup classes I didn't use spaces. Your call. E.g. EventHubs, ServiceBus, AzureStorage

File: src/ExtensionsSample/Program.cs (hunk 1 above)
Comment (fabiocav) [reply src/ExtensionsSample/Program.cs]: Made the change to simplify that process in https://github.com/Azure/azure-webjobs-sdk/pull/1834

Still need to address the proper host ID provider logic to replace what we had before the migration.

**Comment by fabiocav** [comment]
Updated the SDK references and used the new host ID method to address feedback. Will merge, publish this and pull it all into the Script Host once the build is done.

/cc @paulbatum

**Pull request merged**